server/container/guice/cassandra-rabbitmq-guice/pom.xml

```diff
@@ -156,6 +165,26 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-guice-jmx</artifactId>
         </dependency>
+        <dependency>
```

Comment:
I feel there must be something wrong in the way we assemble servers. 

Shouldn't these dependencies be in a common guice server definition ? (by these dependencies I mean the ones that are not specific to a particular storage technology. 

I had a look and `james-server-guice-common` feels like a good candidate but for some reason it pulls in inmemory implementations of some components  such as `james-server-task-memory` `james-server-mailrepository-memory`

Reply:
>Shouldn't these dependencies be in a common guice server definition ? (by these dependencies I mean the ones that are not specific to a particular storage technology.

I could decide to build a server without a given protocol (say JMAP, IMAP, POP3) for instance, to build a pure MTA.

> I had a look and james-server-guice-common feels like a good candidate but for some reason it pulls in inmemory implementations of some components such as james-server-task-memory james-server-mailrepository-memory

Yes some cleanup still to be done, to be honnest.

We might have considered by the past that inMemory dependencies were light with no extra external dependencies thus having them sneaking in was acceptable.